### PR TITLE
[hail] [bugfix/feature] Fix min/max and NaN

### PR DESCRIPTION
**Pull request by tpoterba**
Added `nanmin` and `nanmax` functions.

This was a headache.

### CODE REVIEW

File: hail/python/hail/expr/functions.py [hunk 1]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
     Note
     ----
-    Missing arguments / array elements are ignored if `filter_missing` is ``True``.
-    If `filter_missing` is ``False``, then any missing element causes the result to
-    be missing.
+    If `filter_missing` is ``True``, then the result is the maximum of
+    non-missing arguments or elements. If `filter_missing` is ``False``, then
+    any missing argument or element causes the result to be missing.
+
+    If any element or argumement is `NaN`, then the result is `NaN`.
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: s/argumement/argument/ (typo appears twice more in this file)

File: hail/python/hail/expr/functions.py [hunk 2]
```diff
@@ -3236,9 +3236,94 @@ def reversed(x):
...
+
+    Examples
+    --------
+
+    Compute the maximum value of an array:
+
+    >>> hl.eval(hl.max([1, 3, 5, 6, 7, 9]))
+    9
+
+    Compute the maximum value of arguments:
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: maybe ought to preserve the previous word choice, "take the (minimum,maximum) value"

File: hail/python/hail/expr/functions.py [hunk 3]
```diff
@@ -3236,9 +3236,94 @@ def reversed(x):
...
+    --------
+
+    Compute the maximum value of an array:
+
+    >>> hl.eval(hl.max([1, 3, 5, 6, 7, 9]))
+    9
+
+    Take the maximum value of arguments:
+
+    >>> hl.eval(hl.max(1.1, 50.1, float('nan')))
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: *hl.nanmax

File: hail/python/hail/expr/functions.py [hunk 4]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
+    --------
+
+    Compute the minimum value of an array:
+
+    >>> hl.eval(hl.min([3, 1, 5, 6, 7, 9]))
+    1
+
+    Take the minimum value of arguments:
+
+    >>> hl.eval(hl.min(1.1, 50.1, float('nan')))
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: *hl.nanmin

File: hail/python/hail/expr/functions.py [hunk 5]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
 
     Take the minimum value of an array:
 
-    >>> hl.eval(hl.min([2, 3, 5, 6, 7, 9]))
-    2
+    >>> hl.eval(hl.min([1, 3, 5, 6, 7, 9]))
+    9
 
-    Take the minimum value:
+    Take the minimum value of values:
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: *of arguments

File: hail/python/hail/expr/functions.py [hunk 6]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
 
-    >>> hl.eval(hl.min(12, 50, 2))
-    2
+    >>> hl.eval(hl.min(1, 50, 2))
+    50
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: = 1

File: hail/python/hail/expr/functions.py [hunk 7]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
 
     Examples
     --------
 
     Take the minimum value of an array:
 
-    >>> hl.eval(hl.min([2, 3, 5, 6, 7, 9]))
-    2
+    >>> hl.eval(hl.min([1, 3, 5, 6, 7, 9]))
+    9
```
Comment (iitalics) [inline hail/python/hail/expr/functions.py]: = 1

File: hail/python/hail/expr/functions.py [hunk 8]
```diff
@@ -3262,54 +3347,97 @@ def max(*exprs, filter_missing: bool = True) -> NumericExpression:
...
+
+    Compute the minimum value of an array:
+
+    >>> hl.eval(hl.nanmin([1.1, 50.1, float('nan')]))
+    1.1
+
+    Take the minimum value of arguments:
+
+    >>> hl.eval(hl.nanmin(1.1, 50.1, float('nan')))
+    1.1at d
```
Comment (johnc1231) [inline hail/python/hail/expr/functions.py]: Tests are failing because of the `at d` here.

**Pull request merged**